binding.cc

```diff
@@ -587,27 +587,22 @@ struct Iterator {
         if (!dbIterator_->Valid()) {
           dbIterator_->SeekToLast();
         } else {
-          std::string keyStr = dbIterator_->key().ToString();
-
-          if (lt_ != NULL) {
-            if (lt_->compare(keyStr) <= 0)
-              dbIterator_->Prev();
-          } else if (lte_ != NULL) {
-            if (lte_->compare(keyStr) < 0)
-              dbIterator_->Prev();
-          } else if (start_ != NULL) {
-            if (start_->compare(keyStr))
-              dbIterator_->Prev();
+          leveldb::Slice key = dbIterator_->key();
+
+          if ((lt_ != NULL && key.compare(*lt_) >= 0) ||
+              (lte_ != NULL && key.compare(*lte_) > 0) ||
+              (start_ != NULL && key.compare(*start_) > 0)) {
+            dbIterator_->Prev();
           }
         }
 
         if (dbIterator_->Valid() && lt_ != NULL) {
-          if (lt_->compare(dbIterator_->key().ToString()) <= 0)
-            dbIterator_->Prev();
+          if (dbIterator_->key().compare(*lt_) >= 0)
+            assert(false);
```

Comment:
@ralphtheninja @peakji Can you think of a case where we hit this code path?

Reply:
I guess this is dead code.

If we're using `start_` correctly, there can be only 3 possible cases for a valid reversed iterator in [L584-L599](https://github.com/Level/leveldown/blob/fb7e970ba3dc68d652b0c4be5a038c71c9ed7c70/binding.cc#L584-L597), so the `key` should never be less than `lt` on [L600](https://github.com/Level/leveldown/blob/fb7e970ba3dc68d652b0c4be5a038c71c9ed7c70/binding.cc#L600):

![seek](https://user-images.githubusercontent.com/219051/67188621-4b89db00-f41f-11e9-99df-c0f29ca4631a.jpg)